src/blocklock/BlocklockSender.sol

```diff
@@ -10,48 +10,97 @@ import {TypesLib} from "../libraries/TypesLib.sol";
 import {BLS} from "../libraries/BLS.sol";
 import {IBlocklockSender} from "../interfaces/IBlocklockSender.sol";
 import {IBlocklockReceiver} from "../interfaces/IBlocklockReceiver.sol";
-import {ISignatureSender} from "../interfaces/ISignatureSender.sol";
-
-import {SignatureReceiverBase} from "../signature-requests/SignatureReceiverBase.sol";
-
-import {DecryptionReceiverBase} from "../decryption-requests/DecryptionReceiverBase.sol";
 import {IDecryptionSender} from "../interfaces/IDecryptionSender.sol";
 
-import {console} from "forge-std/console.sol";
-
+import {DecryptionReceiverBase} from "../decryption-requests/DecryptionReceiverBase.sol";
+import {BlocklockFeeCollector} from "./BlocklockFeeCollector.sol";
+
+/// @title BlocklockSender Contract
+/// @author Randamu
+/// @notice This contract is responsible for managing the blocklock sending functionality,
+///         including handling requests, decryption keys, decryption, fees, and access control.
+/// @dev The contract integrates multiple functionalities including decryption receiver capabilities,
+///      fee collection, and role-based access control. It is also upgradeable and follows the UUPS pattern.
+///      The contract implements the `IBlocklockSender` interface and uses `DecryptionReceiverBase` for handling decryption logic.
+///      Additionally, it collects fees via `BlocklockFeeCollector` and uses OpenZeppelin's upgradeable and access control mechanisms.
 contract BlocklockSender is
     IBlocklockSender,
     DecryptionReceiverBase,
+    BlocklockFeeCollector,
     Initializable,
     UUPSUpgradeable,
     AccessControlEnumerableUpgradeable
 {
+    /// @notice This contract manages blocklock requests, decryption keys, and administrative roles.
+    /// @dev The contract includes constants related to blocklock schemes, decryption key processing, and events for blocklock requests and callbacks.
+    ///      It also defines an `ADMIN_ROLE` for managing access control and updates to decryption sender.
+
+    /// @notice The role identifier for the admin role used for access control
+    /// @dev This constant is derived from the keccak256 hash of the string "ADMIN_ROLE" and is used in access control checks
```

Comment:
should check with @azixus how best to embed chainID in any of these DSTs if necessary

Reply:
np shouldn't need them I think